### DISCUSSION TITLE
Can the Service Provider (SP) detect if the user is still logged in to the Identity Provider (IDP)? I want the SP to log out the user if they are signed out on the IDP (like Keycloak)

### DISCUSSION THREAD

**Discussion by maivanhung12c1** [keycloak/keycloak#30689] [Q&A]
Hello everyone,

I'm currently implementing OIDC and SAML with Keycloak for my project. Can the Service Provider (SP) detect whether a user is still logged in to the Identity Provider (IDP)? I want the SP to log out the user if they are signed out on the IDP (like Keycloak).

Thanks.

**Comment by danielFesenmeyer** [comment]
Probably you can use backchannel-logout. Here's an article how to achieve this with Spring Boot, but you can apply the idea to any language/framework: https://developers.redhat.com/articles/2022/12/07/how-implement-single-sign-out-keycloak-spring-boot

**Reply by maivanhung12c1** [reply]
If I configure using backchannel-logout as shown below, when I click on the sign-out button on the session page, will Keycloak send a logout request to 'https://10.22.10.10:443/user/logout'?
![backchannel-logout](https://github.com/keycloak/keycloak/assets/48687271/cd3db4a3-c502-47e2-be01-e3f9bc0db559)
![session](https://github.com/keycloak/keycloak/assets/48687271/0f468fd8-2b46-4dc4-9298-62fe9200fe20)

**Reply by danielFesenmeyer** [reply]
Not completely sure, but according to this discussion, it should work: https://keycloak.discourse.group/t/backchannel-logout-url-not-working/17121

**Reply by maivanhung12c1** [reply]
Thank you very much. Keycloak did not send a request to my web application because I used a self-signed SSL certificate. I added the certificates to conf/truststore and everything worked as expected. However, does SAML support sending a request upon sign out like OIDC?
